src/robusta/core/discovery/discovery.py

```
@@ -184,6 +207,91 @@ def create_service_info_from_hikaru(obj: Union[Deployment, DaemonSet, StatefulSe
+                    limit=DISCOVERY_BATCH_SIZE,
+                    _continue=continue_ref,
+                )
+                items.extend(crd_res.get("items", []))
+                continue_ref = crd_res.get("metadata", {}).get("continue")
+                if not continue_ref:
+                    break
+            except ApiException as e:
+                if e.status == 403:
+                    raise ResourceAccessForbiddenError(
+                        f"Access forbidden to resource '{kind}' in group '{api_group}': {e.body}"
+                    )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)